proctor-common/src/test/java/com/indeed/proctor/common/TestPayloadType.java

```diff
@@ -72,13 +74,16 @@ public void testPayloadTypeForValueRetrieval_StringArrayType() {
     }
 
     @Test
-    public void testPayloadTypeForValueRetrieval_NonArrayTypes() {
+    public void testPayloadTypeForValueRetrieval_NonArrayTypes() throws IOException {
         assertEquals(PayloadType.MAP, payloadTypeForValue(ImmutableMap.of("string", "string")));
         assertEquals(PayloadType.LONG_VALUE, payloadTypeForValue(100));
         assertEquals(PayloadType.LONG_VALUE, payloadTypeForValue(100L));
         assertEquals(PayloadType.DOUBLE_VALUE, payloadTypeForValue(1.1D));
         assertEquals(PayloadType.DOUBLE_VALUE, payloadTypeForValue(2.1F));
         assertEquals(PayloadType.STRING_VALUE, payloadTypeForValue("yes"));
+
+        final ObjectMapper objectMapper = new ObjectMapper();
+        assertEquals(PayloadType.JSON, payloadTypeForValue(objectMapper.readTree(" { \"key\" :  \"value\" } ")));
```

Comment:
Isn't the parsing function implemented in this MR?

Reply:
This function in particular that is being tested does not do any parsing it just takes in the the value that would be in the Payload and checks if it is an instance of any valid Payload Types. 

Could you elaborate on this comment with further details on what you are looking for. Thank you!

Reply:
I imagined we need some parsing function to use this JSON type.
Are you going to implement it or we already have it?

Reply:
There is an additional MR for the internal-webapp that will allow reading in the JSON payloads. From my understanding the Javascript to Java parsing is handled in that project and then other parsing is just using the serializer library.